stunnel/stunnel.ml

```diff
@@ -272,13 +272,15 @@ let rec retry f = function
     deleted.  Instead, it is the caller's responsibility to delete it.  This
     allows the caller to use diagnose_failure below if stunnel fails.  *)
 let with_connect
+    ?(verify_cert=false)
     ?unique_id
     ?use_fork_exec_helper
     ?write_to_log
     ?(extended_diagnosis=false)
     host
     port f = 
-  let _verify_cert = !verify_tls_certs in
+  (* CP-35584 HACK *)
+  let _verify_cert = !verify_tls_certs || verify_cert in
```

Comment:
@psafont we don't need to make your suggested change in xapi (this one https://github.com/xapi-project/xen-api/pull/4278#discussion_r540357190) because we already do this logical or'ing here